Composer/packages/client/src/components/ManageService/ManageService.tsx

```diff
@@ -0,0 +1,702 @@
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { SubscriptionClient } from '@azure/arm-subscriptions';
+import { Subscription } from '@azure/arm-subscriptions/esm/models';
+import { TokenCredentials } from '@azure/ms-rest-js';
+import { CognitiveServicesManagementClient } from '@azure/arm-cognitiveservices';
+import { ResourceManagementClient } from '@azure/arm-resources';
+import { ChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
+import { ProvisionHandoff } from '@bfc/ui-shared';
+import sortBy from 'lodash/sortBy';
+import { NeutralColors } from '@uifabric/fluent-theme';
+
+import { AuthClient } from '../../utils/authClient';
+import { AuthDialog } from '../../components/Auth/AuthDialog';
+import { armScopes } from '../../constants';
+import { getTokenFromCache, isShowAuthDialog, userShouldProvideTokens } from '../../utils/auth';
+import { dispatcherState } from '../../recoilModel/atoms';
+
+type ManageServiceProps = {
+  createService: (
+    tokenCredentials: TokenCredentials,
+    subscriptionId: string,
+    resourceGroupName: string,
+    resourceName: string,
+    region: string
+  ) => Promise<string>;
+  createServiceInBackground?: boolean;
+  handoffInstructions: string;
+  hidden: boolean;
+  learnMore?: string;
+  onDismiss: () => void;
+  onGetKey: (settings: { key: string; region: string }) => void;
+  onNext?: () => void;
+  serviceName: string;
+  regions?: IDropdownOption[];
+  serviceKeyType: string;
+  setVisible: (value: any) => void;
+};
+
+type KeyRec = {
+  name: string;
+  region: string;
+  resourceGroup: string;
+  key: string;
+};
+
+const dropdownStyles = { dropdown: { width: '100%', marginBottom: 20 } };
+const summaryLabelStyles = { display: 'block', color: '#605E5C', fontSize: 14 };
+const summaryStyles = { background: '#F3F2F1', padding: '1px 1rem' };
+const mainElementStyle = { marginBottom: 20 };
+const dialogBodyStyles = { height: 360 };
+const CREATE_NEW_KEY = 'CREATE_NEW';
+
+export const ManageService = (props: ManageServiceProps) => {
+  const [showAuthDialog, setShowAuthDialog] = useState(false);
+  const [token, setToken] = useState<string | undefined>();
+
+  const { setApplicationLevelError } = useRecoilValue(dispatcherState);
+  const [subscriptionId, setSubscription] = useState<string>('');
+  const [resourceGroups, setResourceGroups] = useState<any[]>([]);
+  const [createResourceGroup, setCreateResourceGroup] = useState<boolean>(false);
+  const [newResourceGroupName, setNewResourceGroupName] = useState<string>('');
+  const [resourceGroupKey, setResourceGroupKey] = useState<string>('');
+  const [resourceGroup, setResourceGroup] = useState<string>('');
+
+  const [showHandoff, setShowHandoff] = useState<boolean>(false);
+  const [resourceName, setResourceName] = useState<string>('');
+  const [loading, setLoading] = useState<string | undefined>(undefined);
+  const [noKeys, setNoKeys] = useState<boolean>(false);
```

Comment:
can you track this with ```keys.length``` state? do you really need an extra state for it?

Reply:
if you decided to keep it, shouldn't it be true by default?

Reply:
No because empty because it hasn't loaded is different than empty because there really are no keys.